v2/tests/log_sink_etw_int/log_sink_etw_int.c

```diff
@@ -0,0 +1,683 @@
+    POOR_MANS_ASSERT(strcmp(parsed_events[0].content, "test_critical") == 0);
+    POOR_MANS_ASSERT(strcmp(parsed_events[0].file, __FILE__) == 0);
+    POOR_MANS_ASSERT(strcmp(parsed_events[0].func, __FUNCTION__) == 0);
+    POOR_MANS_ASSERT(parsed_events[0].line == captured_line);
+
+    // error event
+    POOR_MANS_ASSERT(strcmp(parsed_events[1].event_name, "LogError") == 0);
+    POOR_MANS_ASSERT(strcmp(parsed_events[1].content, "test_error") == 0);
+    POOR_MANS_ASSERT(strcmp(parsed_events[1].file, __FILE__) == 0);
+    POOR_MANS_ASSERT(strcmp(parsed_events[1].func, __FUNCTION__) == 0);
+    POOR_MANS_ASSERT(parsed_events[1].line == captured_line + 1);
+
+    // warning event
+    POOR_MANS_ASSERT(strcmp(parsed_events[2].event_name, "LogWarning") == 0);
+    POOR_MANS_ASSERT(strcmp(parsed_events[2].content, "test_warning") == 0);
+    POOR_MANS_ASSERT(strcmp(parsed_events[2].file, __FILE__) == 0);
+    POOR_MANS_ASSERT(strcmp(parsed_events[2].func, __FUNCTION__) == 0);
+    POOR_MANS_ASSERT(parsed_events[2].line == captured_line + 2);
+
+    // info event
+    POOR_MANS_ASSERT(strcmp(parsed_events[3].event_name, "LogInfo") == 0);
+    POOR_MANS_ASSERT(strcmp(parsed_events[3].content, "test_info") == 0);
+    POOR_MANS_ASSERT(strcmp(parsed_events[3].file, __FILE__) == 0);
+    POOR_MANS_ASSERT(strcmp(parsed_events[3].func, __FUNCTION__) == 0);
+    POOR_MANS_ASSERT(parsed_events[3].line == captured_line + 3);
+
+    // verbose event
+    POOR_MANS_ASSERT(strcmp(parsed_events[4].event_name, "LogVerbose") == 0);
+    POOR_MANS_ASSERT(strcmp(parsed_events[4].content, "test_verbose") == 0);
+    POOR_MANS_ASSERT(strcmp(parsed_events[4].file, __FILE__) == 0);
+    POOR_MANS_ASSERT(strcmp(parsed_events[4].func, __FUNCTION__) == 0);
+    POOR_MANS_ASSERT(parsed_events[4].line == captured_line + 4);
+}
+
+static void log_sink_etw_log_each_individual_level(void)
+{
+    // arrange
+    for (LOG_LEVEL event_log_level = LOG_LEVEL_CRITICAL; event_log_level <= LOG_LEVEL_VERBOSE; event_log_level++)
+    {
+        for (LOG_LEVEL trace_level = TRACE_LEVEL_CRITICAL; trace_level <= TRACE_LEVEL_VERBOSE; trace_level++)
+        {
+            generate_trace_session();
+
+            TRACEHANDLE trace_session_handle = start_trace(trace_level);
+
+            int captured_line = __LINE__;
+            uint8_t expected_event_count = (trace_level - TRACE_LEVEL_CRITICAL) >= (event_log_level - LOG_LEVEL_CRITICAL) ? 1 : 0;
+
+            start_parse_events();
+
+            // act
+            log_sink_etw.log_sink_log(event_log_level, NULL, __FILE__, __FUNCTION__, captured_line, "test_event");
+
+            // assert
+            wait_for_event_count(expected_event_count);
+
+            // if we are not expecting an event, wait for some time just to make sure no event is seen
+            if (expected_event_count == 0)
+            {
+                (void)printf("Waiting for %.02f to make sure no events are published ...", (double)TEST_WAITTIME_NO_EVENT / 1000);
```

Comment:
>%.02f [](http://example.com/codeflow?start=42&length=5)

this should indicate what the wait is measured in (in this case, seems to be seconds).

Reply:
Added